src/main/com/github/soerxpso/portalfix/PortalFix.java

```diff
@@ -160,6 +164,40 @@ public void run() {
 		}
 	}
 	
+	class Countdown extends BukkitRunnable {
+		int timeLeft, timeBetweenMessages;
+		Player p;
+		Countdown child = null;
+		
+		//timeLeft and timeBetweenMessages should be in seconds
+		public Countdown(int timeLeft, int timeBetweenMessages, Player p) {
+			this.timeLeft = timeLeft;
+			this.timeBetweenMessages = timeBetweenMessages;
+			this.p = p;
+		}
+		
+		public void run() {
+			p.sendMessage(doColors(Config.getCountdownMessage().replace("%0", String.valueOf(timeLeft / 20))));
+			if(timeLeft - timeBetweenMessages > 0) {
+				child = new Countdown(timeLeft - timeBetweenMessages, timeBetweenMessages, p);
+				child.runTaskLater(PortalFix.getPlugin(), timeBetweenMessages);
+			}
+		}
+		
+		//Devoted is not responsible for the possible results of taking life advice from method names
+		public void killSelfAndChildren() {
+			if(child == null) {
+				this.cancel();
+			}else {
+				child.killSelfAndChildren();
+			}
+		}
+	}
```

Comment:
Using the runTaskTimer should let you avoid this method. Although it is clever :P